packages/eslint-plugin-next/src/index.ts

```diff
@@ -1,4 +1,50 @@
-const recommendedRules = {
+import type { ESLint, Linter, Rule } from 'eslint'
+
+const rules: Record<string, Rule.RuleModule> = {
+  'google-font-display':
+    require('./rules/google-font-display') as typeof import('./rules/google-font-display'),
+  'google-font-preconnect':
+    require('./rules/google-font-preconnect') as typeof import('./rules/google-font-preconnect'),
+  'inline-script-id':
+    require('./rules/inline-script-id') as typeof import('./rules/inline-script-id'),
+  'next-script-for-ga':
+    require('./rules/next-script-for-ga') as typeof import('./rules/next-script-for-ga'),
+  'no-assign-module-variable':
+    require('./rules/no-assign-module-variable') as typeof import('./rules/no-assign-module-variable'),
+  'no-async-client-component':
+    require('./rules/no-async-client-component') as typeof import('./rules/no-async-client-component'),
+  'no-before-interactive-script-outside-document':
+    require('./rules/no-before-interactive-script-outside-document') as typeof import('./rules/no-before-interactive-script-outside-document'),
+  'no-css-tags':
+    require('./rules/no-css-tags') as typeof import('./rules/no-css-tags'),
+  'no-document-import-in-page':
+    require('./rules/no-document-import-in-page') as typeof import('./rules/no-document-import-in-page'),
+  'no-duplicate-head':
+    require('./rules/no-duplicate-head') as typeof import('./rules/no-duplicate-head'),
+  'no-head-element':
+    require('./rules/no-head-element') as typeof import('./rules/no-head-element'),
+  'no-head-import-in-document':
+    require('./rules/no-head-import-in-document') as typeof import('./rules/no-head-import-in-document'),
+  'no-html-link-for-pages':
+    require('./rules/no-html-link-for-pages') as typeof import('./rules/no-html-link-for-pages'),
+  'no-img-element':
+    require('./rules/no-img-element') as typeof import('./rules/no-img-element'),
+  'no-page-custom-font':
+    require('./rules/no-page-custom-font') as typeof import('./rules/no-page-custom-font'),
+  'no-script-component-in-head':
+    require('./rules/no-script-component-in-head') as typeof import('./rules/no-script-component-in-head'),
+  'no-styled-jsx-in-document':
+    require('./rules/no-styled-jsx-in-document') as typeof import('./rules/no-styled-jsx-in-document'),
+  'no-sync-scripts':
+    require('./rules/no-sync-scripts') as typeof import('./rules/no-sync-scripts'),
+  'no-title-in-document-head':
+    require('./rules/no-title-in-document-head') as typeof import('./rules/no-title-in-document-head'),
+  'no-typos': require('./rules/no-typos') as typeof import('./rules/no-typos'),
+  'no-unwanted-polyfillio':
+    require('./rules/no-unwanted-polyfillio') as typeof import('./rules/no-unwanted-polyfillio'),
+}
+
+const recommendedRules: Linter.RulesRecord = {
```

Comment:
this doesnt preserve the rule names, I had created a pr earlier https://github.com/vercel/next.js/pull/82202 that fixes the typing issue while also preserving the rule names in the object

Reply:
Ah, good catch! I'll switch to using `satisfies` here.